lib/index.js

```diff
@@ -48,27 +48,46 @@ var Drip = module.exports = integration('Drip')
  * @api public
  */
 
-Drip.prototype.identify = function(identify, fn){
+Drip.prototype.identify = function(identify, fn) {
   var settings = this.settings;
   var account = settings.account;
   var url = fmt('%s/subscribers', account);
   var subscriber = mapper.identify(identify);
   var self = this;
 
-  this
-    .post(url)
-    .auth(settings.token)
-    .type('json')
-    .send({ subscribers: [subscriber] })
-    .end(function(err, res){
-      if (err && err.timeout) return fn(err);
-      if (res.ok) return fn(null, res);
-      var errors = res.body.errors || [{}];
-      var msg = errors[0].message;
-      var status = res.status;
-      if ('Email is already subscribed' == msg) return fn(null, res);
-      fn(self.error('bad request status=%s msg=%s', status, msg || ''));
-    });
+  var req = function() {
+    self.post(url)
+        .auth(settings.token)
+        .type('json')
+        .send({subscribers: [subscriber]})
+        .end(function(err, res) {
+          if (err && err.timeout) {
+            return fn(err)
+          }
+
+          if (res.ok) {
+            return self.setLimit(fn(null, res));
+          }
+
+          var errors = res.body.errors || [{}];
+          var msg = errors[0].message;
+          var status = res.status;
+
+          if (res.status == 429) {
+            err = new Error('too many requests')
+            err.status = 429;
+            return self.setLimit(fn(err));
+          }
+
+          if ('Email is already subscribed' == msg) {
+            return fn(null, res);
+          }
+
+          fn(self.error('bad request status=%s msg=%s', status, msg || ''));
```

Comment:
I think we're probably better off just passing `err` here instead of `self.error(...)`. This will pass the error code out properly and the error message will still be available in the logs